src/nvim/eval.c

```diff
@@ -15255,8 +15278,9 @@ static void f_setloclist(typval_T *argvars, typval_T *rettv)
   rettv->vval.v_number = -1;
 
   win = find_win_by_nr(&argvars[0], NULL);
-  if (win != NULL)
-    set_qf_ll_list(win, &argvars[1], &argvars[2], rettv);
+  if (win != NULL) {
+    set_qf_ll_list(win, &argvars[1], &argvars[2], &argvars[3], rettv);
```

Comment:
> Perhaps it would be easier if we change the arguments of set_qf_ll_list() and just forward the argument array

:+1: If you're not already doing it, I'll look into it later today.